### ISSUE TITLE
Throws an error if one of the arguments dosen't have a name

### ISSUE THREAD

**Issue by Keno** [cstjean/Unrolled.jl#6]
```
julia> @unroll function sim_gbm(state, sim_T, drift, vol, ts::TimeStep, ::Val{log_scale}) where log_scale
           log_scale && (state = log.(state))
           @unroll for _ in 1:length(ts)
               if log_scale
                   state = sim_gbm_euler_step(state, drift, vol_matrix, dt(ts))
               else
                   state = sim_gbm_log_scale_euler_step(state, drift, vol_matrix, dt(ts))
               end
           end
           return log_scale ? exp.(state) : state
       end
ERROR: LoadError: AssertionError: Bad ::Val{log_scale}
Stacktrace:
 [1] macro expansion at /home/keno/.julia/packages/MacroTools/4AjBS/src/macro.jl:73 [inlined]
 [2] function_argument_name(::Expr) at /home/keno/.julia/dev/Unrolled/src/Unrolled.jl:71
 [3] iterate at ./generator.jl:47 [inlined]
 [4] collect_to!(::Array{Symbol,1}, ::Base.Generator{Array{Any,1},typeof(Unrolled.function_argument_name)}, ::Int64, ::Int64) at ./array.jl:656
 [5] collect_to_with_first! at ./array.jl:643 [inlined]
 [6] _collect(::Array{Any,1}, ::Base.Generator{Array{Any,1},typeof(Unrolled.function_argument_name)}, ::Base.EltypeUnknown, ::Base.HasShape{1}) at ./array.jl:637
 [7] collect_similar(::Array{Any,1}, ::Base.Generator{Array{Any,1},typeof(Unrolled.function_argument_name)}) at ./array.jl:561
 [8] map(::Function, ::Array{Any,1}) at ./abstractarray.jl:2014
 [9] @unroll(::LineNumberNode, ::Module, ::Any) at /home/keno/.julia/dev/Unrolled/src/Unrolled.jl:88
in expression starting at REPL[10]:1
```

**Comment by cstjean** [comment]
Fixed by [d5bde5c](/cstjean/Unrolled.jl/commit/d5bde5cd253401f73df57cca92d163b8a23d793b)

**Comment by Keno** [comment]
Thanks!
